qa/common/nfs-ganesha.sh

```diff
@@ -0,0 +1,105 @@
+#
+# This file is part of the DeepSea integration test suite
+#
+
+function _nfs_ganesha_node {
+  _first_x_node ganesha
+}
+
+function nfs_ganesha_no_root_squash {
+  local GANESHAJ2=/srv/salt/ceph/ganesha/files/ganesha.conf.j2
+  sed -i '/Access_Type = RW;/a \\tSquash = No_root_squash;' $GANESHAJ2
+}
+
+#
+# Since we don't seem to be using NFSv4, the effect of this option is unclear
+#
+function nfs_ganesha_no_grace_period {
+  local GANESHAJ2=/srv/salt/ceph/ganesha/files/ganesha.conf.j2
+  cat <<EOF >>$GANESHAJ2
+NFSv4 {Graceless = True}
+EOF
+}
+
+function nfs_ganesha_debug_log {
+  local GANESHANODE=$(_nfs_ganesha_node)
+  local TESTSCRIPT=/tmp/test-nfs-ganesha.sh
+  cat <<EOF > $TESTSCRIPT
+set -ex
+trap 'echo "Result: NOT_OK"' ERR
+echo "nfs-ganesha debug log script running as $(whoami) on $(hostname --fqdn)"
+sed -i 's/NIV_EVENT/NIV_DEBUG/g' /etc/sysconfig/nfs-ganesha
+cat /etc/sysconfig/nfs-ganesha
+rm -rf /var/log/ganesha/ganesha.log
+systemctl restart nfs-ganesha.service
+systemctl is-active nfs-ganesha.service
+echo "Result: OK"
+EOF
+  _run_test_script_on_node $TESTSCRIPT $GANESHANODE
+}
+
+function nfs_ganesha_cat_config_file {
+  salt -C 'I@roles:ganesha' cmd.run 'cat /etc/ganesha/ganesha.conf'
+}
+
+function nfs_ganesha_showmount_loop {
+  local TESTSCRIPT=/tmp/test-nfs-ganesha.sh
+  salt -C 'I@roles:ganesha' cmd.run "while true ; do showmount -e $GANESHANODE | tee /tmp/showmount.log || true ; grep -q 'Timed out' /tmp/showmount.log || break ; done"
```

Comment:
I have exactly the same issue. Everytime I restart the nfs-ganesha service when a client issues the first command, either showmounts or mount, it will return a timeout. I also find out that if I just kill the command before getting the timeout and re-issue the command again, it works. So maybe this is actually a bug of the nfs-ganesha service.